### PR TITLE
Add workflow for cleaning up a single workspace [sc-51778]

### PR DESCRIPTION
**Pull request by e3b0c442**
#### What this PR does / why we need it:
Adds a reusable workflow for cleaning up a single e2e test workspace. This will eventually be called by a workflow which finds the workspaces ready to be cleaned up (and can also be called as one-off where needed).

### CODE REVIEW

File: .github/workflows/e2e-workspace-cleanup.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+          repository: replicatedhq/kots-regression-automation
+          token: ${{ secrets.E2E_GH_PAT }}
+          path: automation
+          ref: main
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v2
+      - name: Initialize Terraform
+        env:
+          TF_WORKSPACE: default
+        run: terraform init -reconfigure -backend-config="key=jumpbox.tfstate"
```
Comment (syntastical) [inline .github/workflows/e2e-workspace-cleanup.yaml:96]: key is already hard coded in the terraform so you shouldn't need it here

File: .github/workflows/e2e-workspace-cleanup.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        with:
+          repository: replicatedhq/kots-regression-automation
+          token: ${{ secrets.E2E_GH_PAT }}
+          path: automation
+          ref: main
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v2
+      - name: Initialize Terraform
+        env:
+          TF_WORKSPACE: default
```
Comment (syntastical) [inline .github/workflows/e2e-workspace-cleanup.yaml:95]: This is hard coded in a couple places, is this just for testing?

**Review by syntastical — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e-workspace-cleanup.yaml (hunk 1 above)
Comment (e3b0c442) [reply .github/workflows/e2e-workspace-cleanup.yaml]: Removed, thanks

File: .github/workflows/e2e-workspace-cleanup.yaml (hunk 2 above)
Comment (e3b0c442) [reply .github/workflows/e2e-workspace-cleanup.yaml:98]: This overrides line 33 which is set by the parameter passed to the workflow

**Pull request merged**